staging/src/k8s.io/component-base/logs/json/json_test.go

```diff
@@ -72,13 +77,26 @@ func TestZapLoggerInfo(t *testing.T) {
 		writer.Flush()
 		logStr := buffer.String()
 		var ts float64
-		n, err := fmt.Sscanf(logStr, data.format, &ts)
-		if n != 1 || err != nil {
-			t.Errorf("log format error: %d elements, error %s:\n%s", n, err, logStr)
-		}
-		expect := fmt.Sprintf(data.format, ts)
-		if !assert.Equal(t, expect, logStr) {
-			t.Errorf("Info has wrong format \n expect:%s\n got:%s", expect, logStr)
+		var lineNo int
+		switch data.noOfFormatArgs {
```

Comment:
Can you split the logStr by `\n` and analyse each line separatly?

Reply:
Oh that's a good idea, let me do that